iothub/device/samples/PnpDeviceSamples/TemperatureController/ComponentTemperatureControllerSampleOld.cs

```diff
@@ -50,20 +48,23 @@ public async Task PerformOperationsAsync(CancellationToken cancellationToken)
                 Humidity = 20,
                 Temperature = 25
             };
+
             if (!properties.Properties.Reported.Contains(Thermostat2)
-                || ((JObject)properties.Properties.Reported[Thermostat2]).Value<ThermostatInitialValue>("initialValue") != initialValue)
+                || !((JObject)properties.Properties.Reported[Thermostat2]).TryGetValue("initialValue", out JToken initialValueToken)
+                || !initialValue.Equals(System.Text.Json.JsonSerializer.Deserialize<ThermostatInitialValue>(initialValueToken.ToString())))
             {
-                var propertiesToBeUpdated = new TwinCollection
+                var propertiesToBeUpdated = new Dictionary<string, object>
                 {
                     ["__t"] = "c",
                     ["initialValue"] = initialValue
                 };
-                var componentProperty = new TwinCollection
+                var componentProperty = new Dictionary<string, object>
                 {
                     [Thermostat2] = propertiesToBeUpdated
                 };
-                await _deviceClient.UpdateReportedPropertiesAsync(propertiesToBeUpdated, cancellationToken);
-                _logger.LogDebug($"Property: Update - {propertiesToBeUpdated.ToJson()}.");
+                var twinCollection = new TwinCollection(System.Text.Json.JsonSerializer.Serialize(componentProperty));
```

Comment:
twin collection will inherently try to serialize the object based on newtonsoft. Since our type uses System.Text.Json, we will create the TwinCollection using the serialized string directly.